packages/api/src/folder/service.ts

```diff
@@ -0,0 +1,96 @@
+import createError from "http-errors";
+import { logger } from "@stela/logger";
+import { db } from "../database";
+import type {
+  FolderColumnsForUpdate,
+  Folder,
+  PatchFolderRequest,
+} from "./models";
+import { requestFieldsToDatabaseFields } from "./helper";
+import type { GetAccountArchiveResult } from "../account/models";
+
+export const getFolderById = async (requestQuery: {
+  folderId: string;
+}): Promise<Folder> => {
+  const folder = await db
+    .sql<Folder>("folder.queries.get_folder_by_id", {
+      folderId: requestQuery.folderId,
+    })
+    .catch((err) => {
+      logger.error(err);
+      throw new createError.InternalServerError("Failed to retrieve folder");
+    });
+
+  if (folder.rows[0] === undefined) {
+    throw new createError.NotFound("Folder not found.");
+  }
+
+  return folder.rows[0];
+};
+
+function buildPatchQuery(patchFolderRequest: FolderColumnsForUpdate): string {
+  const updates = Object.entries(patchFolderRequest)
+    .filter(([key, value]) => value !== undefined && key !== "folderId")
+    .map(([key, _]) => `${key} = :${key}`);
+
+  if (updates.length === 0) {
+    throw new createError.BadRequest("Request cannot be empty");
+  }
+
+  const query = `
+    UPDATE folder
+    SET ${updates.join(", ")}
+    WHERE folderid = :folderId
+    RETURNING folder.folderid AS "folderId"
+  `;
+
+  return query.trim();
+}
+
+const canPatchFolder = async (
```

Comment:
We also need to check the role of the account in the relevant archive (https://permanent.zohodesk.com/portal/en/kb/articles/roles-for-collaboration-and-sharing), and it's possible that it has access via an entry in the `access` table (which represents shares) rather than the `account_archive` table.

Reply:
Ok, I'm not 100% of my solution to this. I see that folders can have multiple folder links and the access info is based on folder link id not on folder id.